src/types.ts

```diff
@@ -1,5 +1,179 @@
-export type Types = {
-  StrategyOptions: any;
-  AuthOptions: any;
-  Authentication: any;
+import * as OctokitTypes from "@octokit/types";
+
+// # Generic Types
+
+// Most types and interfaces are generic around two types:
+//
+// 1. App type: If it’s an OAuth App or a GitHub App.
+// 2. Expiration setting: If token expiration has been enabled for the app.
+//
+// Although currently GitHub does not support token expiration for OAuth app,
+// expiring OAuth tokens will become the norm in future. These two concepts are
+// modeled orthogonally.
+export type ClientType = OAuthApp | GitHubApp; // TODO: Better named `AppType`
```

Comment:
I know the naming is confusing, but because of the context of OAuth's nomenclature of `client_id`/`client_secret` I thought it would make more sense to use `ClientType`. In the context of the OAuth flow specification, the server that sends the request to GitHub's OAuth endpoints is the client, hence "client type".

We use it in other places and shoudl be consistent.